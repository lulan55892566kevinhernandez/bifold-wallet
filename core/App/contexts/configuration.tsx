import { createContext, useContext } from 'react'

import { RecordProps } from '../components/record/Record'
import OnboardingPages from '../screens/OnboardingPages'
import { Overlay } from '../types/overlay'

export interface ConfigurationContext {
  pages: typeof OnboardingPages
  splash: React.FC
  terms: React.FC
  homeContentView: React.FC
  OCABundle: { branding: Record<string, Overlay>; oca: JSON | undefined }
  useBiometry: React.FC
  record: React.FC<RecordProps>
}

export const ConfigurationContext = createContext<ConfigurationContext>(null as unknown as ConfigurationContext)

export const ConfigurationProvider = ConfigurationContext.Provider

export const useConfiguration = () => useContext(ConfigurationContext)
