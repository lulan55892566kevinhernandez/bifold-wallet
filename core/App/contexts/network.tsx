import { NetInfoStateType, useNetInfo } from '@react-native-community/netinfo'
import * as React from 'react'
import { createContext, useContext, useState } from 'react'

import NetInfoModal from '../components/modals/NetInfoModal'

export interface NetworkContext {
  silentAssertConnectedNetwork: () => boolean
  assertConnectedNetwork: () => boolean
  displayNetInfoModal: () => void
  hideNetInfoModal: () => void
}

export const NetworkContext = createContext<NetworkContext>(null as unknown as NetworkContext)

export const NetworkProvider: React.FC = ({ children }) => {
  const netInfo = useNetInfo()
  const [isNetInfoModalDisplayed, setIsNetInfoModalDisplayed] = useState<boolean>(false)

  const displayNetInfoModal = () => {
    setIsNetInfoModalDisplayed(true)
  }

  const hideNetInfoModal = () => {
    setIsNetInfoModalDisplayed(false)
  }

  const silentAssertConnectedNetwork = () => {
    return netInfo.isConnected || netInfo.type !== NetInfoStateType.none
  }

  const assertConnectedNetwork = () => {
    const isConnected = silentAssertConnectedNetwork()
    if (!isConnected) {
      displayNetInfoModal()
    }
    return isConnected
  }

  return (
    <NetworkContext.Provider
      value={{
        silentAssertConnectedNetwork,
        assertConnectedNetwork,
        displayNetInfoModal,
        hideNetInfoModal,
      }}
    >
      {children}
      <NetInfoModal visible={isNetInfoModalDisplayed} onSubmit={() => hideNetInfoModal()} />
    </NetworkContext.Provider>
  )
}

export const useNetwork = () => useContext(NetworkContext)
