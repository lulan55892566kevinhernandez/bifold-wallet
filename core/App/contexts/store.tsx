import React, { createContext, Dispatch, useContext, useReducer } from 'react'

import { State } from '../types/state'

import reducer, { ReducerAction } from './reducers/store'

const initialState: State = {
  onboarding: {
    didAgreeToTerms: false,
    didCompleteTutorial: false,
    didCreatePIN: false,
    didConsiderBiometry: false,
  },
  authentication: {
    didAuthenticate: false,
  },
  lockout: {
    displayNotification: false,
  },
  privacy: {
    didShowCameraDisclosure: false,
  },
  preferences: {
    useBiometry: false,
  },
  credential: {
    revoked: new Set(),
    revokedMessageDismissed: new Set(),
  },
  error: null,
  loading: false,
}

export const StoreContext = createContext<[State, Dispatch<ReducerAction>]>([
  initialState,
  () => {
    return
  },
])

export const StoreProvider: React.FC = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState)
  return <StoreContext.Provider value={[state, dispatch]}>{children}</StoreContext.Provider>
}

export const useStore = () => useContext(StoreContext)
