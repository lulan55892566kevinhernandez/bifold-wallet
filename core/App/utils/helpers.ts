import {
  Agent,
  ConnectionRecord,
  CredentialExchangeRecord,
  RequestedAttribute,
  RequestedPredicate,
  RetrievedCredentials,
  IndyCredentialInfo,
  IndyProofFormat,
} from '@aries-framework/core'
import {
  FormatDataMessagePayload,
  FormatRetrievedCredentialOptions,
} from '@aries-framework/core/build/modules/proofs/models/ProofServiceOptions'
import { useConnectionById } from '@aries-framework/react-hooks'
import { Buffer } from 'buffer'
import moment from 'moment'
import { parseUrl } from 'query-string'

import { i18n } from '../localization/index'
import { ProofCredentialAttributes, ProofCredentialPredicates } from '../types/record'

export { parsedCredDefName } from './cred-def'
export { parsedSchema } from './schema'

/**
 * Generates a numerical hash based on a given string
 * @see https://stackoverflow.com/questions/3426404/create-a-hexadecimal-colour-based-on-a-string-with-javascript
 * @param { string } s given string
 * @returns { number } numerical hash value
 */
export const hashCode = (s: string): number => {
  return s.split('').reduce((hash, char) => char.charCodeAt(0) + ((hash << 5) - hash), 0)
}

/**
 * Generates a pseudorandom number between 0 and 1 based on a seed
 * @see https://gist.github.com/tommyettinger/46a874533244883189143505d203312c
 * @see https://stackoverflow.com/questions/521295/seeding-the-random-number-generator-in-javascript
 * @param { number } seed any number
 * @returns { number } pseudorandom number between 0 and 1
 */
const mulberry32 = (seed: number) => {
  let t = (seed += 0x6d2b79f5)
  t = Math.imul(t ^ (t >>> 15), t | 1)
  t ^= t + Math.imul(t ^ (t >>> 7), t | 61)
  return ((t ^ (t >>> 14)) >>> 0) / 4294967296
}

/**
 * Converts a numerical hash into a hexidecimal color string
 * @see https://helderesteves.com/generating-random-colors-js/#Generating_random_dark_colors
 * @param { number } hash numerical hash value (generated by hashCode function above)
 * @returns { string } hexidecimal string eg. #32d3cc
 */
export const hashToRGBA = (hash: number) => {
  let color = '#'
  const colorRangeUpperBound = 256

  // once for r, g, b
  for (let i = 0; i < 3; i++) {
    // append a pseudorandom two-char hexidecimal value from the lower half of the color spectrum (to limit to darker colors)
    color += ('0' + Math.floor((mulberry32(hash + i) * colorRangeUpperBound) / 2).toString(16)).slice(-2)
  }

  return color
}

export function formatTime(time: Date, params?: { long?: boolean; format?: string }): string {
  const getMonthKey = 'MMMM'
  const momentTime = moment(time)
  const monthKey = momentTime.format(getMonthKey)
  const customMonthFormatRe = /M+/
  const long = params?.long
  const format = params?.format
  const shortDateFormatMaskLength = 3

  let formatString = i18n.t('Date.ShortFormat')
  if (format) {
    formatString = format
  } else {
    if (long) {
      formatString = i18n.t('Date.LongFormat')
    }

    // if translation fails
    if (formatString === 'Date.ShortFormat' || formatString === 'Date.LongFormat' || formatString === undefined) {
      formatString = 'MMM D, YYYY'
    }
  }
  const customMonthFormat = formatString?.match(customMonthFormatRe)?.[0]

  let formattedTime = momentTime.format(formatString)

  if (customMonthFormat) {
    let monthReplacement = ''
    const monthReplacementKey = momentTime.format(customMonthFormat)
    if (customMonthFormat.length === shortDateFormatMaskLength) {
      // then we know we're dealing with a short date format: 'MMM'
      monthReplacement = i18n.t(`Date.MonthShort.${monthKey}`)
    } else if (customMonthFormat.length > shortDateFormatMaskLength) {
      // then we know we're working with a long date format: 'MMMM'
      monthReplacement = i18n.t(`Date.MonthLong.${monthKey}`)
    }
    // if translation doesn't work
    if (monthReplacement === `Date.MonthLong.${monthKey}` || monthReplacement === `Date.MonthShort.${monthKey}`) {
      monthReplacement = monthReplacementKey
    }

    if (monthReplacement) {
      formattedTime = formattedTime.replace(monthReplacementKey, monthReplacement)
    }
  }
  return formattedTime
}

/**
 * @deprecated The function should not be used
 */
export function connectionRecordFromId(connectionId?: string): ConnectionRecord | void {
  if (connectionId) {
    return useConnectionById(connectionId)
  }
}

/**
 * @deprecated The function should not be used
 */
export function getConnectionName(connection: ConnectionRecord | void): string | void {
  if (!connection) {
    return
  }
  return connection?.alias || connection?.theirLabel
}

export function getCredentialConnectionLabel(credential?: CredentialExchangeRecord) {
  if (!credential) {
    return ''
  }

  return credential?.connectionId
    ? useConnectionById(credential.connectionId)?.theirLabel
    : credential?.connectionId ?? ''
}

export function getConnectionImageUrl(connectionId: string) {
  const connection = useConnectionById(connectionId)
  if (!connection) {
    return undefined
  }
  return connection.imageUrl ?? undefined
}

export function firstValidCredential(
  fields: RequestedAttribute[] | RequestedPredicate[],
  revoked = true
): RequestedAttribute | RequestedPredicate | null {
  if (!fields.length) {
    return null
  }

  let first = null
  const firstNonRevoked = fields.filter((field) => !field.revoked)[0]
  if (firstNonRevoked) {
    first = firstNonRevoked
  } else if (fields.length && revoked) {
    first = fields[0]
  }

  if (!first?.credentialInfo) {
    return null
  }

  return first
}

export const getOobDeepLink = async (url: string, agent: Agent | undefined): Promise<any> => {
  const queryParams = parseUrl(url).query
  const b64Message = queryParams['d_m'] ?? queryParams['c_i']
  const rawmessage = Buffer.from(b64Message as string, 'base64').toString()
  const message = JSON.parse(rawmessage)
  await agent?.receiveMessage(message)
  return message
}

/**
 * A sorting function for the Array `sort()` function
 * @param a First retrieved credential
 * @param b Second retrieved credential
 */
export const credentialSortFn = (a: any, b: any) => {
  if (a.revoked && !b.revoked) {
    return 1
  } else if (!a.revoked && b.revoked) {
    return -1
  } else {
    return b.timestamp - a.timestamp
  }
}

export const processProofAttributes = (
  request?: FormatDataMessagePayload<[IndyProofFormat], 'request'> | undefined,
  credentials?: FormatRetrievedCredentialOptions<[IndyProofFormat]>
): { [key: string]: ProofCredentialAttributes } => {
  const processedAttributes = {} as { [key: string]: ProofCredentialAttributes }

  if (!(request?.indy?.requested_attributes && credentials?.proofFormats?.indy?.requestedAttributes)) {
    return {}
  }

  const requestedProofAttributes = request.indy.requested_attributes
  const retrievedCredentialAttributes = credentials.proofFormats.indy.requestedAttributes

  for (const key of Object.keys(retrievedCredentialAttributes)) {
    // The shift operation modifies the original input array, therefore make a copy
    const credential = [...(retrievedCredentialAttributes[key] ?? [])].sort(credentialSortFn).shift()

    const credName = key
    let revoked = false
    if (credential) {
      revoked = credential.revoked as boolean
    }
    const { name, names } = requestedProofAttributes[key]

    for (const attributeName of [...(names ?? (name && [name]) ?? [])]) {
      if (!processedAttributes[credName]) {
        // init processedAttributes object
        processedAttributes[credName] = {
          schemaId: credential?.credentialInfo?.schemaId,
          credDefId: credential?.credentialInfo?.credentialDefinitionId,
          credName,
          attributes: [],
        }
      }

      let attributeValue = '' //(credentialInfo as IndyCredentialInfo).attributes[attributeName]
      if (credential) {
        attributeValue = (credential.credentialInfo as IndyCredentialInfo).attributes[attributeName]
      }
      processedAttributes[credName].attributes?.push({
        revoked,
        name: attributeName,
        value: attributeValue,
      })
    }
  }
  return processedAttributes
}

export const processProofPredicates = (
  request?: FormatDataMessagePayload<[IndyProofFormat], 'request'> | undefined,
  credentials?: FormatRetrievedCredentialOptions<[IndyProofFormat]>
): { [key: string]: ProofCredentialPredicates } => {
  const processedPredicates = {} as { [key: string]: ProofCredentialPredicates }

  if (!(request?.indy?.requested_predicates && credentials?.proofFormats?.indy?.requestedPredicates)) {
    return {}
  }

  const requestedProofPredicates = request.indy.requested_predicates
  const retrievedCredentialPredicates = credentials.proofFormats.indy.requestedPredicates

  for (const key of Object.keys(requestedProofPredicates)) {
    // The shift operation modifies the original input array, therefore make a copy
    const credential = [...(retrievedCredentialPredicates[key] ?? [])].sort(credentialSortFn).shift()
    const { credentialId, revoked, credentialDefinitionId, schemaId } = { ...credential, ...credential?.credentialInfo }
    const { name, p_type: pType, p_value: pValue } = requestedProofPredicates[key]

    const credName = key

    if (!processedPredicates[credName]) {
      processedPredicates[credName] = {
        schemaId,
        credDefId: credentialDefinitionId,
        credName: credName,
        predicates: [],
      }
    }

    processedPredicates[credName].predicates?.push({
      credentialId,
      name,
      revoked,
      pValue,
      pType,
    })
  }
  return processedPredicates
}

/**
 * @deprecated The function should not be used
 */
export const sortCredentialsForAutoSelect = (credentials: RetrievedCredentials): RetrievedCredentials => {
  const requestedAttributes = Object.values(credentials?.requestedAttributes).pop()
  const requestedPredicates = Object.values(credentials?.requestedPredicates).pop()
  const sortFn = (a: any, b: any) => {
    if (a.revoked && !b.revoked) {
      return 1
    } else if (!a.revoked && b.revoked) {
      return -1
    } else {
      return b.timestamp - a.timestamp
    }
  }

  requestedAttributes && requestedAttributes.sort(sortFn)
  requestedPredicates && requestedPredicates.sort(sortFn)

  return credentials
}

/**
 *
 * @param url a redirection URL to retrieve a payload for an invite
 * @param agent an Agent instance
 * @returns payload from following the redirection
 */
export const receiveMessageFromUrlRedirect = async (url: string, agent: Agent | undefined) => {
  const res = await fetch(url, {
    method: 'GET',
    headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
  })
  const message = await res.json()
  await agent?.receiveMessage(message)
  return message
}

/**
 *
 * @param url a redirection URL to retrieve a payload for an invite
 * @param agent an Agent instance
 * @returns payload from following the redirection
 */
export const receiveMessageFromDeepLink = async (url: string, agent: Agent | undefined) => {
  const res = await fetch(url, {
    method: 'GET',
    headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
  })
  const message = await res.json()
  await agent?.receiveMessage(message)
  return message
}

/**
 *
 * @param uri a URI containing a base64 encoded connection invite in the query parameter
 * @param agent an Agent instance
 * @returns a connection record from parsing and receiving the invitation
 */
export const connectFromInvitation = async (uri: string, agent: Agent | undefined) => {
  const invitation = await agent?.oob.parseInvitation(uri)

  if (!invitation) {
    throw new Error('Could not parse invitation from URL')
  }

  const record = await agent?.oob.receiveInvitation(invitation)
  const connectionRecord = record?.connectionRecord
  if (!connectionRecord?.id) {
    throw new Error('Connection does not have an ID')
  }

  return connectionRecord
}
