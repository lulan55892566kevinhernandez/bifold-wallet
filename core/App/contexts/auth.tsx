import React, { createContext, useContext, useState } from 'react'

import { secretForPIN, storeWalletSecret, loadWalletSecret, convertToUseBiometrics } from '../services/keychain'
import { WalletSecret } from '../types/security'
import { hashPIN } from '../utils/crypto'

export interface AuthContext {
  checkPIN: (pin: string) => Promise<boolean>
  convertToUseBiometrics: () => Promise<boolean>
  getWalletCredentials: () => Promise<WalletSecret | undefined>
  setPIN: (pin: string) => Promise<void>
}

export const AuthContext = createContext<AuthContext>(null as unknown as AuthContext)

export const AuthProvider: React.FC = ({ children }) => {
  const [walletSecret, setWalletSecret] = useState<WalletSecret>()

  const setPIN = async (pin: string): Promise<void> => {
    const secret = await secretForPIN(pin)
    await storeWalletSecret(secret)
  }

  const checkPIN = async (pin: string): Promise<boolean> => {
    const secret = await loadWalletSecret()

    if (!secret || !secret.salt) {
      return false
    }

    const hash = await hashPIN(pin, secret?.salt)

    return hash === secret.walletKey
  }

  const getWalletCredentials = async (): Promise<WalletSecret | undefined> => {
    if (walletSecret) {
      return walletSecret
    }

    const secret = await loadWalletSecret()
    if (!secret || !secret.walletKey) {
      return
    }

    delete secret.salt
    setWalletSecret(secret)

    return secret
  }

  return (
    <AuthContext.Provider
      value={{
        checkPIN,
        convertToUseBiometrics,
        getWalletCredentials,
        setPIN,
      }}
    >
      {children}
    </AuthContext.Provider>
  )
}

export const useAuth = () => useContext(AuthContext)
