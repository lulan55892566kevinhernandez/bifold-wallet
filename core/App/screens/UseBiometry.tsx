import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { StyleSheet, Text, View, Switch, StatusBar, Platform } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'

import Biometrics from '../assets/img/biometrics.svg'
import Button, { ButtonType } from '../components/buttons/Button'
import { useAuth } from '../contexts/auth'
import { DispatchAction } from '../contexts/reducers/store'
import { useStore } from '../contexts/store'
import { useTheme } from '../contexts/theme'
import { statusBarStyleForColor, StatusBarStyles } from '../utils/luminance'
import { testIdWithKey } from '../utils/testable'

const UseBiometry: React.FC = () => {
  const [, dispatch] = useStore()
  const { t } = useTranslation()
  const { convertToUseBiometrics } = useAuth()
  const [biometryEnabled, setBiometryEnabled] = useState(false)
  const [continueEnabled, setContinueEnabled] = useState(true)
  const { ColorPallet, TextTheme } = useTheme()
  const styles = StyleSheet.create({
    container: {
      flexGrow: 2,
      flexDirection: 'column',
      paddingHorizontal: 25,
      backgroundColor: ColorPallet.brand.primaryBackground,
    },
    image: {
      minWidth: 200,
      minHeight: 200,
      marginBottom: 66,
    },
  })

  const continueTouched = async () => {
    setContinueEnabled(false)

    if (biometryEnabled) {
      await convertToUseBiometrics()
    }

    dispatch({
      type: DispatchAction.USE_BIOMETRY,
      payload: [biometryEnabled],
    })
  }

  const toggleSwitch = () => setBiometryEnabled((previousState) => !previousState)

  return (
    <SafeAreaView style={styles.container}>
      <StatusBar
        barStyle={Platform.OS === 'android' ? StatusBarStyles.Light : statusBarStyleForColor(ColorPallet.brand.primary)}
      />
      <View style={{ flexGrow: 1 }}>
        <View style={{ alignItems: 'center' }}>
          <Biometrics style={[styles.image]} />
        </View>
        <Text style={[TextTheme.normal]}>{t('Biometry.Text1')}</Text>
        <Text></Text>
        <Text style={[TextTheme.normal]}>
          {t('Biometry.Text2')}
          <Text style={[TextTheme.normal, { fontWeight: 'bold' }]}> {t('Biometry.Warning')}</Text>
        </Text>
        <View
          style={{
            flexDirection: 'row',
            marginVertical: 30,
          }}
        >
          <View style={{ flexShrink: 1 }}>
            <Text style={[TextTheme.normal, { fontWeight: 'bold' }]}>{t('Biometry.UseToUnlock')}</Text>
          </View>
          <View style={{ justifyContent: 'center' }}>
            <Switch
              accessibilityLabel={t('Biometry.Toggle')}
              testID={testIdWithKey('ToggleBiometrics')}
              trackColor={{ false: ColorPallet.grayscale.lightGrey, true: ColorPallet.brand.primaryDisabled }}
              thumbColor={biometryEnabled ? ColorPallet.brand.primary : ColorPallet.grayscale.mediumGrey}
              ios_backgroundColor={ColorPallet.grayscale.lightGrey}
              onValueChange={toggleSwitch}
              value={biometryEnabled}
            />
          </View>
        </View>
        <View style={{ flexGrow: 1, justifyContent: 'flex-end' }}>
          <Button
            title={'Continue'}
            accessibilityLabel={'Continue'}
            testID={testIdWithKey('Continue')}
            onPress={continueTouched}
            buttonType={ButtonType.Primary}
            disabled={!continueEnabled}
          />
        </View>
      </View>
    </SafeAreaView>
  )
}

export default UseBiometry
