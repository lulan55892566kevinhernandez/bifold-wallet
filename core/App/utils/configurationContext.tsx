import { createContext, useContext } from 'react'

import { pages } from '../screens/OnboardingPages'

export interface ConfigurationContext {
  onboarding: {
    pages: typeof pages
  }
  splash: React.FC
  terms: React.FC
}

const ConfigurationContext = createContext<ConfigurationContext>(null as unknown as ConfigurationContext)
export const ConfigurationProvider = ConfigurationContext.Provider

export const useConfigurationContext = () => useContext(ConfigurationContext)
