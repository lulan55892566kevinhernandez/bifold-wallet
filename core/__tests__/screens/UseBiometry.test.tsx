import { render, fireEvent, waitFor } from '@testing-library/react-native'
import React from 'react'

import { AuthContext } from '../../App/contexts/auth'
import UseBiometry from '../../App/screens/UseBiometry'
import { testIdWithKey } from '../../App/utils/testable'

jest.mock('@react-navigation/core', () => {
  return require('../../__mocks__/custom/@react-navigation/core')
})
jest.mock('@react-navigation/native', () => {
  return require('../../__mocks__/custom/@react-navigation/native')
})

jest.mock('react-native-device-info', () => {
  return {
    isEmulatorSync: () => {
      return true
    },
  }
})

describe('UseBiometry Screen', () => {
  test.skip('Renders correctly when biometry available', () => {
    // const tree = render(
    //   <AuthContext.Provider
    //     value={{
    //       checkPIN: jest.fn(),
    //       commitPIN: jest.fn(),
    //       getWalletCredentials: jest.fn(),
    //       setPIN: jest.fn(),
    //       isBiometricsActive: jest.fn().mockReturnValue(Promise.resolve(true)),
    //     }}
    //   >
    //     <UseBiometry />
    //   </AuthContext.Provider>
    // )
    expect(tree).toMatchSnapshot()
  })

  test.skip('Renders correctly when biometry not available', () => {
    // const tree = render(
    //   <AuthContext.Provider
    //     value={{
    //       checkPIN: jest.fn(),
    //       commitPIN: jest.fn(),
    //       getWalletCredentials: jest.fn(),
    //       setPIN: jest.fn(),
    //       isBiometricsActive: jest.fn().mockReturnValue(Promise.resolve(false)),
    //     }}
    //   >
    //     <UseBiometry />
    //   </AuthContext.Provider>
    // )
    expect(tree).toMatchSnapshot()
  })

  test.skip('Toggles use biometrics ok', async () => {
    const commitPIN = jest.fn()
    // const tree = render(
    //   <AuthContext.Provider
    //     value={{
    //       checkPIN: jest.fn(),
    //       commitPIN,
    //       getWalletCredentials: jest.fn(),
    //       setPIN: jest.fn(),
    //       isBiometricsActive: jest.fn().mockReturnValue(Promise.resolve(true)),
    //     }}
    //   >
    //     <UseBiometry />
    //   </AuthContext.Provider>
    // )

    // const useBiometryToggle = await tree.getByTestId(testIdWithKey('ToggleBiometrics'))
    // await waitFor(async () => {
    //   await fireEvent(useBiometryToggle, 'valueChange', true)
    // })

    // const continueButton = await tree.getByTestId(testIdWithKey('Continue'))
    // await waitFor(async () => {
    //   await fireEvent(continueButton, 'press')
    // })

    // expect(useBiometryToggle).not.toBeNull()
    // expect(continueButton).not.toBeNull()
    // expect(commitPIN).toBeCalledTimes(1)
    expect(tree).toMatchSnapshot()
  })
})
