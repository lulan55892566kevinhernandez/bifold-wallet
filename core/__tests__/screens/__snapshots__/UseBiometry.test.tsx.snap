// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UseBiometry Screen Renders correctly 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "#000000",
      "flexDirection": "column",
      "flexGrow": 2,
      "paddingHorizontal": 25,
    }
  }
>
  <View
    style={
      Object {
        "flexGrow": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <
        style={
          Array [
            Object {
              "marginBottom": 66,
              "minHeight": 200,
              "minWidth": 200,
            },
          ]
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "normal",
          },
        ]
      }
    >
      Biometry.Text1
    </Text>
    <Text />
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "normal",
          },
        ]
      }
    >
      Biometry.Text2
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "bold",
            },
          ]
        }
      >
         
        Biometry.Warning
      </Text>
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginVertical": 30,
        }
      }
    >
      <View
        style={
          Object {
            "flexShrink": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "bold",
              },
            ]
          }
        >
          Biometry.UseToUnlock
        </Text>
      </View>
      <View
        style={
          Object {
            "justifyContent": "center",
          }
        }
      >
        <RCTSwitch
          accessibilityLabel="Biometry.Toggle"
          accessibilityRole="switch"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onTintColor="rgba(53, 130, 63, 0.35)"
          style={
            Array [
              Object {
                "height": 31,
                "width": 51,
              },
              Object {
                "backgroundColor": "#D3D3D3",
                "borderRadius": 16,
              },
            ]
          }
          testID="com.ariesbifold:id/ToggleBiometrics"
          thumbTintColor="#606060"
          tintColor="#D3D3D3"
          value={false}
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flexGrow": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        accessibilityLabel="Continue"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#42803E",
            "borderRadius": 4,
            "opacity": 1,
            "padding": 16,
          }
        }
        testID="com.ariesbifold:id/Continue"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              false,
            ]
          }
        >
          Continue
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`UseBiometry Screen Toggles use biometrics ok 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "#000000",
      "flexDirection": "column",
      "flexGrow": 2,
      "paddingHorizontal": 25,
    }
  }
>
  <View
    style={
      Object {
        "flexGrow": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <
        style={
          Array [
            Object {
              "marginBottom": 66,
              "minHeight": 200,
              "minWidth": 200,
            },
          ]
        }
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "normal",
          },
        ]
      }
    >
      Biometry.Text1
    </Text>
    <Text />
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "normal",
          },
        ]
      }
    >
      Biometry.Text2
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 18,
              "fontWeight": "normal",
            },
            Object {
              "fontWeight": "bold",
            },
          ]
        }
      >
         
        Biometry.Warning
      </Text>
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginVertical": 30,
        }
      }
    >
      <View
        style={
          Object {
            "flexShrink": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "bold",
              },
            ]
          }
        >
          Biometry.UseToUnlock
        </Text>
      </View>
      <View
        style={
          Object {
            "justifyContent": "center",
          }
        }
      >
        <RCTSwitch
          accessibilityLabel="Biometry.Toggle"
          accessibilityRole="switch"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onTintColor="rgba(53, 130, 63, 0.35)"
          style={
            Array [
              Object {
                "height": 31,
                "width": 51,
              },
              Object {
                "backgroundColor": "#D3D3D3",
                "borderRadius": 16,
              },
            ]
          }
          testID="com.ariesbifold:id/ToggleBiometrics"
          thumbTintColor="#42803E"
          tintColor="#D3D3D3"
          value={true}
        />
      </View>
    </View>
    <View
      style={
        Object {
          "flexGrow": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        accessibilityLabel="Continue"
        accessibilityState={
          Object {
            "disabled": true,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(53, 130, 63, 0.35)",
            "borderRadius": 4,
            "opacity": 1,
            "padding": 16,
          }
        }
        testID="com.ariesbifold:id/Continue"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "bold",
                "textAlign": "center",
              },
            ]
          }
        >
          Continue
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
